### PR TITLE
Pio namelist fix

### PR DESCRIPTION
**Pull request by jedwards4b**
The pio_numtasks wasn't being set correctly.  After each external call to a component buildnml we must update the case object to assure that changes made by the external are captured.

Test suite: scripts_regression_tests, GECO case in cesm
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes

User interface changes?: 

Code review:

### CODE REVIEW

File: cime_config/cesm/machines/config_compilers.xml [hunk 1]
```diff
@@ -978,7 +978,9 @@ using a fortran linker.
   <FFLAGS>
     <append> -vec-report </append>
   </FFLAGS>
-
+  <CMAKE_OPTS>
```
Comment (jgfouca) [inline cime_config/cesm/machines/config_compilers.xml]: Does ACME need these XML changes?

File: cime_config/cesm/machines/config_machines.xml [hunk 2]
```diff
@@ -956,6 +956,7 @@
     </module_system>
     <environment_variables>
       <env name="OMP_STACKSIZE">256M</env>
+      <env name="MPI_TYPE_DEPTH">16</env>
```
Comment (jgfouca) [inline cime_config/cesm/machines/config_machines.xml]: Ditto ^

Comment (jedwards4b) [inline cime_config/cesm/machines/config_machines.xml]: No - these are specific to the new ncar machine laramie

**Comment by jgfouca** [comment]
I approve... will wait for @mvertens

**Pull request merged**